### PR TITLE
Support wallet providers other than web3onboard

### PR DESCRIPTION
**Pull request by sstefdev**
Fixes https://github.com/RequestNetwork/web-components/issues/31

Changes:
- Added rainbowkit and connect button
- Added wagmi config and provider
- Changed props for the web components
- Added styling for the connect button
- Update Web Components versions

## Release Notes

- **New Features**
	- Introduced a new `ConnectButton` component for improved wallet connection management.
	- Added comprehensive deployment instructions for Vercel, EasyPanel, and GitHub Actions.
	- New section in the README for environment variables to aid in deployment.

- **Improvements**
	- Restructured the README for better clarity on configuration and deployment processes.
	- Enhanced the `Navbar` component by integrating the new `ConnectButton`, simplifying wallet connection logic.
	- Updated wallet integration configuration to utilize the `wagmi` library, improving modularity and user experience.

- **Dependency Updates**
	- Updated several dependencies, including `@requestnetwork/create-invoice-form` and `@requestnetwork/invoice-dashboard`, while removing outdated packages related to wallet connections.

- **Documentation**
	- Expanded and clarified the README to enhance usability for developers.

### CODE REVIEW

File: components/common/ConnectButton.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                    className={`flex items-center gap-[10px] disabled:cursor-not-allowed disabled:bg-grey disabled:text-dark-grey text-white rounded-[8px] hover:bg-dark-green transition-all text-[14px] tablet:px-[28px]  px-[14px] lg:px-[20px] text-14px lg:text-[16px] py-[8px] bg-green focus:outline-none`}
+                  >
+                    Connect Wallet
+                  </button>
+                );
+              }
+              if (chain.unsupported) {
+                return (
+                  <button onClick={openChainModal} type="button">
+                    Wrong network
```
Comment (MantisClone) [inline components/common/ConnectButton.tsx:50]: Can we say "Unsupported network" instead?

File: pages/_app.tsx [hunk 2]
```diff
@@ -1,36 +1,37 @@
...
-              "https://github.com/RequestNetwork/invoicing-template/releases"
-            }
-          />
-        </Provider>
-      </Web3OnboardProvider>
+      <WagmiProvider config={rainbowKitConfig}>
+        <QueryClientProvider client={queryClient}>
+          <RainbowKitProvider>
+            <Provider>
+              <Navbar /> <Component {...pageProps} />
```
Comment (MantisClone) [inline pages/_app.tsx:25]: Can you move these to separate lines?

File: utils/adaptWalletProvider.ts
Comment (MantisClone) [inline utils/adaptWalletProvider.ts:1]: We don't need this file. Please remove.

File: types.d.ts [hunk 3]
```diff
@@ -21,14 +21,14 @@ interface CreateInvoiceFormElement {
 
 interface InvoiceDashboardProps extends HTMLElement {
   config: IConfig;
-  wallet: WalletState;
+  wagmiConfig: WagmiConfig;
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply types.d.ts:24]: Readme updated

File: package.json [hunk 4]
```diff
@@ -11,20 +11,14 @@
...
-    "@web3-onboard/core": "^2.21.6",
-    "@web3-onboard/gnosis": "^2.2.2",
-    "@web3-onboard/injected-wallets": "^2.10.16",
-    "@web3-onboard/ledger": "^2.6.0",
-    "@web3-onboard/react": "^2.9.3",
-    "@web3-onboard/trezor": "^2.4.4",
-    "@web3-onboard/trust": "^2.0.4",
-    "@web3-onboard/walletconnect": "^2.5.5",
+    "@tanstack/react-query": "^5.59.14",
+    "@wagmi/core": "^2.13.8",
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply package.json:21]: Package-lock updated

File: components/common/ConnectButton.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        openConnectModal,
+        authenticationStatus,
+        mounted,
+      }) => {
+        const ready = mounted && authenticationStatus !== "loading";
+        const connected =
+          ready &&
+          account &&
+          chain &&
+          (!authenticationStatus || authenticationStatus === "authenticated");
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply components/common/ConnectButton.tsx:23]: No need for changing

File: components/common/ConnectButton.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            {...(!ready && {
+              "aria-hidden": true,
+              style: {
+                opacity: 0,
+                pointerEvents: "none",
+                userSelect: "none",
+              },
+            })}
+          >
+            {(() => {
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply components/common/ConnectButton.tsx:35]: Too much if statements, we can just use css

File: pages/_app.tsx [hunk 7]
```diff
@@ -1,36 +1,38 @@
...
+              <Component {...pageProps} />
+              <VersionDisplay
+                githubRelease={
+                  "https://github.com/RequestNetwork/invoicing-template/releases"
+                }
+              />
+            </Provider>
+          </RainbowKitProvider>
+        </QueryClientProvider>
+      </WagmiProvider>
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply pages/_app.tsx:35]: Wrong, Tanstack query Provider should be the child of WagmiProvider:
https://wagmi.sh/react/getting-started#setup-tanstack-query

**Review by MantisClone — APPROVED** [review]
Approved contingent on all comments being resolved.

**Pull request merged**